File: SkillBridgeSmsService/src/main/java/com/skillBridge/sms_service/service/AdminService.java
package com.skillBridge.sms_service.service;

import java.util.List;

import com.skillBridge.sms_service.dtos.AdminRegisterRequest;
import com.skillBridge.sms_service.dtos.ProjectResponse;
import com.skillBridge.sms_service.dtos.StudentResponse;
import com.skillBridge.sms_service.entities.ProjectStatus;
import com.skillBridge.sms_service.entities.Role;

public interface AdminService {

    StudentResponse create(AdminRegisterRequest request);

    List<StudentResponse> getAll();

    StudentResponse getById(Long id);

    void updateRole(Long id, Role role);

    void delete(Long id);
    
    //=========PROJECT ADMIN ===============
    List<ProjectResponse> getAllProjects();

    ProjectResponse updateProjectStatus(Long projectId, ProjectStatus status);

    void deleteProject(Long projectId);

	
}

File: SkillBridgeFrontend/src/services/adminService.js
import api from './api';

export const adminService = {
    // Get all students
    getAllStudents: async () => {
        const response = await api.get('/admin/students');
        return response.data;
    },

    // Get student by ID
    getStudentById: async (id) => {
        const response = await api.get(`/admin/students/${id}`);
        return response.data;
    },

    // Update student role
    updateRole: async (id, role) => {
        const response = await api.patch(`/admin/students/${id}/role?role=${role}`);
        return response.data;
    },

    // Delete student
    deleteStudent: async (id) => {
        const response = await api.delete(`/admin/students/${id}`);
        return response.data;
    },

    // Get all projects
    getAllProjects: async () => {
        const response = await api.get('/admin/projects');
        return response.data;
    },

    // Update project status
    updateProjectStatus: async (id, status) => {
        const response = await api.patch(`/admin/projects/${id}/status?status=${status}`);
        return response.data;
    },

    // Delete project
    deleteProject: async (id) => {
        const response = await api.delete(`/admin/projects/${id}`);
        return response.data;
    },
};

export default adminService;
